Merging SSMProblems.jl into this repo

shravanngoswamii opened the pull request:
Continues #29, Let me know if I you want me to add Research directory in root folder!

I used git subtree to merge `SSMProblems.jl` in this repo, here are the steps:
```sh
git remote add ssm https://github.com/TuringLang/SSMProblems.jl.git
git fetch ssm
git subtree add --prefix=SSMProblems ssm main
```
This command imports the entire commit history of SSMProblems.jl and places its files under the SSMProblems folder in current repository.

yebai committed:
Initial commit

FredericWantiez committed:
Readme

FredericWantiez committed:
Project

yebai committed:
Merge pull request #1 from TuringLang/feature/readme

Feature/readme

yebai committed:
Prototype design.  (#2)

* prototype

* Update SSMProblems.jl

* add logM (#3)

* Convert example into docstring.

* Move `logM`.

---------

Co-authored-by: FredericWantiez <frederic.wantiez@gmail.com>

FredericWantiez committed:
export

FredericWantiez committed:
example

FredericWantiez committed:
Fred/ancestor (#5)

* Gibbs

* Add ancestor resampling

* Better names

FredericWantiez committed:
Clean up package (#6)

* Gibbs

* Setup github

* Update Readme

* Docs

* GH actions

* Format

* Upgrade node

* Use GH token

* Fix links

* Clean up

FredericWantiez committed:
Write a proper example implementation (#7)

* SMC

* remove old file

* Fix types

* Apply suggestions from code review

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

---------

Co-authored-by: Hong Ge <3279477+yebai@users.noreply.github.com>
Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

torfjelde committed:
Some minor changes (#8)

* use recommended style for interface methods

* minor changes to example

* Update smc.jl

* Update smc.jl

---------

Co-authored-by: Hong Ge <3279477+yebai@users.noreply.github.com>

THargreaves committed:
Particle filter example bug fix and refactoring (#10)

* fix: corrected observation generation for
particle filter example

* refactor: tidied particle filter example code

- Removed recursive particle show method which flooded REPL
- Removed redundant resampling logic
- Replaced variance with std in Normal() calls
- Tidied final scatter plot

* Updated formatting for named argument

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

* fix: corrected flipped noise standard deviations

---------

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

devmotion committed:
Fix format action (#13)

devmotion committed:
Fix docs action (#12)

* Fix docs action

* Add DOCUMENTER_KEY

* Use julia-docdeploy action

devmotion committed:
Show link to docs preview (#15)

FredericWantiez committed:
Update documentation (#16)

* Add details to doc

* Fix source

* Typo

* Update SSM Interface

* Fix linearize bug

* Interface

* Format

* Trying things

* Fix transition!!

* Format

* Fix doc

* Helper

* Format

* Forget about particles

* Optional timestep

* Add utils

* Apply suggestions from code review

Co-authored-by: David Widmann <devmotion@users.noreply.github.com>

* Utils module

---------

Co-authored-by: David Widmann <devmotion@users.noreply.github.com>

yebai committed:
Update README.md (#17)

* Update README.md

* Update README.md

yebai committed:
Update README.md

yebai committed:
Update README.md

yebai committed:
Update README.md

yebai committed:
Update README.md

yebai committed:
Update README.md

yebai committed:
Update index.md

THargreaves committed:
Update documentation to match new interface (#18)

* Update documentation to match new interface

* Update index.md

---------

Co-authored-by: Hong Ge <3279477+yebai@users.noreply.github.com>

FredericWantiez committed:
Build examples with doc (#19)

* Build examples with doc

* Reduce size of plot

* Colors

* Size option

* Increase size per page

yebai committed:
Update README.md

yebai committed:
Update Project.toml

THargreaves committed:
Modify SSMProblems to work with AbstractMCMC interface (#22)

* Update documentation to match new interface

* Incorporate SSMProblems into AbstractMCMC

* Update Project.toml (#24)

* Update Project.toml

* Update make.jl

---------

Co-authored-by: Hong Ge <3279477+yebai@users.noreply.github.com>

THargreaves committed:
Add Kalman filter example (#26)

* Add Kalman filter example

* Fix formatting issues

* Add literate for docs

* add missing deps

* Comments for literate

* Format

* Use `Gaussian` (#28)

* Format, use `Gaussian`

* Fix the maths

* Format

* Tweaks

* Update script.jl

* Update script.jl

---------

Co-authored-by: Hong Ge <3279477+yebai@users.noreply.github.com>

* Update Project.toml

---------

Co-authored-by: FredericWantiez <frederic.wantiez@gmail.com>
Co-authored-by: Hong Ge <hg344@cam.ac.uk>
Co-authored-by: Hong Ge <3279477+yebai@users.noreply.github.com>

yebai committed:
Update script.jl

yebai committed:
Update script.jl

yebai committed:
Create DocsNav.yml

THargreaves committed:
Add example script for PMMH (#37)

* Add example script for PMMH

* Add Literate.jl

yebai committed:
Update script.jl

yebai committed:
Update script.jl

yebai committed:
Update DocsNav.yml

THargreaves committed:
Update SSMProblems.jl interface (#38)

* Add split dynamics/observation interface with "extra" variables

* Add utilities for forward simulation and distribution definitions

* Removed redundant particle container code

* Update naming convention for initialisation log-density

Co-authored-by: Hong Ge <3279477+yebai@users.noreply.github.com>

* Update initialisation naming

Co-authored-by: Hong Ge <3279477+yebai@users.noreply.github.com>

* Update naming convention

Co-authored-by: Hong Ge <3279477+yebai@users.noreply.github.com>

* Change sampler to AbstractMCMC

Co-authored-by: Hong Ge <3279477+yebai@users.noreply.github.com>

* Add section heading for SSM

Co-authored-by: Hong Ge <3279477+yebai@users.noreply.github.com>

* Remove redundant method check

* Correct dependencies

* Revert to positional arguments

* Correct forward simulation element type

* Update Kalman filter example to new interface

* Fix formatting issue

* Add missing import

Co-authored-by: Charles Knipp <32943413+charlesknipp@users.noreply.github.com>

* Add default rngs to samplers through macro

* Remove unnecessary section heading

* Add missing dependency

* Tidied Kalman filter example

* Update documentation main page

* Fully document Kalman filter example

* Remove outdated examples

* Add documentation for extra argument

* Apply suggestions from code review

* Update script.jl

* Update examples/kalman-filter/script.jl

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

* Update script.jl

* Remove PMMH (until new API)

* Remove ref to 'Utils'

* Fix broken link

* Remove default RNG macro

* Simplify interface methods

* Correct old function names

* Add parametric type to Kalman filter

* Update main doc page

* Update README

* Make parameter order consistent

---------

Co-authored-by: Hong Ge <3279477+yebai@users.noreply.github.com>
Co-authored-by: Charles Knipp <32943413+charlesknipp@users.noreply.github.com>
Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: FredericWantiez <frederic.wantiez@gmail.com>

yebai committed:
Minor tweaks and typo fixes  (#41)

* Update index.md

* Update index.md

* Update Project.toml

* Suppress output in example script

---------

Co-authored-by: THargreaves <tim.hargreaves@icloud.com>

THargreaves committed:
Split method definitions to avoid docstring overwriting (#42)

willtebbutt committed:
Function docstring formatting (#45)

shravanngoswamii committed:
added TagBot & CompatHelper workflows (#47)

* added TagBot & CompatHelper workflows

* using existing Documenter Key for CompatHelper

github-actions[bot] committed:
CompatHelper: add new compat entry for Distributions at version 0.25, (keep existing compat) (#48)

Co-authored-by: CompatHelper Julia <compathelper_noreply@julialang.org>

yebai committed:
Update DocsPreviewCleanup.yml

shravanngoswamii committed:
TagBot Permission Issue fixed (#50)

penelopeysm committed:
Add DOCUMENTER_KEY to Docs workflow (#52)

yebai committed:
Update DocsPreviewCleanup.yml

FredericWantiez commented:
Haven't done a proper benchmark but it seems like its helping a bit, but not much.

THargreaves committed:
Correct type signature for forward simulation method

THargreaves committed:
Alignment of obs/dyn time steps and refactored forward simulation (#55)

* Update interface documentation to align dyn/obs time steps

* Refactor forward simulation, add type parameters, add unit test

* Add type parameters to docstrings

* Update kalman example

* Add extra for initialisation and simplify obs type parameter

* Bump minor version

* Update unit tests to match aligned interface

* Fix code comment rendering

charlesknipp committed:
Interface Changes for Use in Filtering (#56)

* added basic particle methods and filters

* added qualifiers

* added parameter priors

* added adaptive resampling to bootstrap filter (WIP)

* Julia fomatter changes

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

* changed eltype for <: StateSpaceModel

* updated naming conventions

* formatter

* fixed adaptive resampling

* added particle ancestry

* formatter issues

* fixed metropolis and added rejection resampler

* Keep track of free indices using stack

* updated particle types and organized directory

* weakened SSM type parameter assertions

* improved particle state containment and resampling

* added hacky sparse ancestry to example

* fixed RNG in rejection resampling

* improved callbacks and resamplers

* formatting

* added conditional SMC

* improved linear model type structure

* formatter

* replaced extra with kwargs

* formatter

* migrated filtering code

* Add unittests for new interface

* Update documentation to match kwargs

* Rename extras/kwargs docs file

* remove redundant forward simulations

---------

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: Tim Hargreaves <tim.hargreaves@icloud.com>

FredericWantiez committed:
Bump 0.4.0 (#58)

THargreaves committed:
Update docs to match kwargs interface

THargreaves committed:
Add method definitions for batch simulation/log-densities

THargreaves committed:
Bump minor version

THargreaves committed:
Correct docstring overwriting for batch methods

THargreaves committed:
Update type parameters to contain both arithmetic and element type

THargreaves committed:
Correct docstring indentations

shravanngoswamii committed:
Documentation and Turing Navigation CI improvement (#61)

* Update Docs.yml

* Update DocsNav.yml

* No need of deploydocs() after using new Docs & DocsNav workflows

shravanngoswamii committed:
Add 'SSMProblems/' from commit '2420413e72b9faa257548818d74e466a083f3dc3'

git-subtree-dir: SSMProblems
git-subtree-mainline: 26d4a029ef611e5887918e932f7da8d0e911a2f5
git-subtree-split: 2420413e72b9faa257548818d74e466a083f3dc3

shravanngoswamii committed:
removed SSMProblems README, LICENSE, GHA workflow, gitignore, JULIAFORMATTER in favor of merger

shravanngoswamii committed:
SSMProblems: added missing docstring to avoid documentation failure

shravanngoswamii committed:
added pkg_path in Docs workflow to fix package development

shravanngoswamii committed:
moved Research to root directory

shravanngoswamii committed:
Merge branch 'main' of https://github.com/TuringLang/GeneralisedFilters.jl into sg/ssm-merge
